ui/desktop/README.md

```diff
@@ -172,23 +174,29 @@ route using `await a11yAudit();`.
 *Be aware this is a temporary process that can suffer changes and eventually will be automated.*
 
 This process will explain how to run end to end test in your local environment. The process will assume next:
+
 - You already run the Boundary UI [installation process](https://github.com/hashicorp/boundary-ui#installation).
 - You have Boundary installed, see [the installation guide](https://learn.hashicorp.com/tutorials/boundary/getting-started-install?in=boundary/getting-started).
 
 Steps:
+
 - Open a terminal and run `$ boundary dev`.
 - Open another terminal and navigate to `$ boundary-ui/ui/desktop` folder.
 - Run `$ yarn run e2e` and tests will start run.
 - You can check the screenshots the tests take: `desktop/tests/end2end/screenshots`.
 
-
 ### Troubleshooting
 
 #### Blank screen and/or hang browser tab when running as web app
 
-We are aware of an issue of the desktop client not being able to start correctly when running as web app within development mode. This issue reproduces when the developer is enabling/disabling Mirage ([more info](#environment-variables) about environment variables).
+We are aware of an issue of the desktop client not being able to start correctly when running as web app within development mode. This issue reproduces when the developer is enabling/disabling Mirage ([more info](#environment-variables-dev) about environment variables).
 
 The workaround to fix this issue is to reset/clear cookies and data (local storage and session storage).
+
+#### Node-gyp build errors after package upgrades
+
+If you are running into issues with running the desktop client after a node upgrade you may need to rebuild some tools needed for building the electron app. Be sure to remove `node_models/` for `ui/desktop/` and `ui/desktop/electron-app/` to make sure you have a clean slate. First thing you will need to check is if you have `python-setuptools` installed. It is needed for node-gyp to rebuild the native tools for electron. If you are on a mac and use Homebrew, you can run `brew install python-setuptools`. You should be able to run `yarn start:desktop` now. This will trigger the rebuild for electron but if you want to be safe you can run `./electron-app/node_modules/.bin/electron-rebuild` before trying to start up the desktop client.
```

Comment:
I would mention this is only for python v3.12 and above as well as the python command to install it if you don't use homebrew

Reply:
This wasn't an issue below python v3.12? Were those tools automatically included with those versions of python?

Reply:
Yup, exactly

Reply:
Updated. Let me know if that works.